src/olympia/addons/utils.py

```diff
@@ -409,3 +415,33 @@ def fetch_translations_from_addon(addon, properties):
     translation_ids = [id_ for id_ in translation_ids_gen if id_]
     # Just get all the values together to make it simplier
     return {str(value) for value in Translation.objects.filter(id__in=translation_ids)}
+
+
+def generate_delete_token(addon_id):
+    token_payload = {
+        'exp': datetime.utcnow() + timedelta(seconds=60),
+        'addon_id': addon_id,
+        'action': 'delete',
+    }
+    return jwt.encode(token_payload, settings.SECRET_KEY)
```

Comment:
I think this was what you had in mind around a special token?

Reply:
Yes, but let's not use `jwt` for this. Use django's `sign`: https://docs.djangoproject.com/en/4.0/topics/signing/

Reply:
done https://github.com/mozilla/addons-server/pull/19147/commits/0597956cbe27508c7cd53a66d89c6886b2d3628a